### ISSUE TITLE
Run Process doesn't work

### ISSUE THREAD

**Issue by Dai-Bang** [synfron/ReshaperForBurp#49]
I tried to run a basic command in Run Process but it doesn't work. Do I have to to put anything in Stdin to make it work? I am using the Burpsuite Pro edition with Linux OS Debian Base zsh. I even try to Run a script but it doesn't work as well. I have the latest version of Burpsuite Pro 10.2

When: 
     Event Direction: Request

Then:
     Run Process: 
         Command: sh -c 'Nordvpn -c'

**Comment by ddwightx** [comment]
@Dai-Bang Thanks for the report. I have confirmed that Run Process does not work well on Linux. I will put on a fix for this via pre-release in the coming days. Full release TBD.

**Comment by Dai-Bang** [comment]
Thank you for your quick respond. I am looking forward for the update. I really like this extension. It has so many features! Thank you for your contribution.

**Comment by ddwightx** [comment]
@Dai-Bang While you're waiting, have you tried the command alone `Nordvpn -c`, basically without the sh part?

**Comment by Dai-Bang** [comment]
I did try that, but it doesn't work either. I even tried it by using your script feature "Run Script" to run a command. It didn't work as well.

**Comment by ddwightx** [comment]
@Dai-Bang Right. Run Script is for running JavaScript, not sh scripts.
 
Another workaround is to create a `whateverName.sh` file in your filesystem with whatever you need to run, then in the Run Process command field, put `sh /full/path/to/whateverName.sh`. If that doesn't work, the problem may not be related to Reshaper. Check that by running the script you created yourself manually to make sure it actually works by itself.

You can right-click the Rule and Toggle Debug Logging or enable full event diagnostics logs in Reshaper on the Settings tab, then watch the Logs tab after the Rule is run to see what the `exitCode` was for the Then Run Process. If the `exitCode` is 0, that's a good sign that the script actually ran.

**Comment by Dai-Bang** [comment]
I tried it and I got exit code 1,2, and sometimes 0. Also, I notice there is a conflict between Reshaper and Logg++. When I tried to run Logg++, Reshaper's rules no longer work. Reshaper's rules only work within Macros. If you execute it in Repeater and Intruder, it doesn't work at all! It took me a longer time to figure out that I have to disable Logg++ extension to make it work.

I see that you just release a new version? Where can I download it. Thanks

**Comment by ddwightx** [comment]
Hi @Dai-Bang, I haven't released a new version yet. Odd, Logger++ should have no effect on Reshaper.

Regarding Repeater and other Burp tools, do you have them enabled in the Reshaper settings tab?

**Comment by Dai-Bang** [comment]
Yes. I did have them enabled in the Reshaper settings tab. I don't know what cause it though. It works fine when you test the Macros but in Repeater and Intruder, it doesn't work. This issue seem to be isolated as well. Some rules are working while others are not?? See example below:

Rule 1:
Whens:  
           Event Direction: Request
           Matches Text: ipinfo.io
Then: 
           Run Process: nordvpn connect
           Delay: 3000

Rule 2:
When: 
          Event Direction: Response
          Matches Text: rdwr
Then:
         Set Even Direction: Response
         Set Value: Response Status Line
         Delete Value: Response Headers
         Delete Value: Response Body

The Rule1 is working fine with Logg++ enabled. For the Rule2, when Logg++ enabled, it's only working inside Testing tab of the Macros. It would not within the Repeater and Intruder. However, Rule 1 is working fine. Therefore, I think there may be some kind of conflict with Logg++, I try to isolate the issue by disabling Rule 1 to test Rule 2. But it does not work either.

I have another question regarding the Send Request.  In Rule 2, I use it to test the validity of the request. I have 3 Macros(1,2,3) running in a session. I added Send Request at the end of Then in Rule 2.  For example, when Macros #2 is running and match Rule 2,  the Request is being sent back to beginning of the Macros #1 instead of just running the current request which is Macros #2??? It create an infinite loop in background. If you look at the front end, the Response will display the result. However, in the Logg, it create an infinite loop running non-stop. I would have to disable my wifi to stop it. It does not break upon failure. Did I do something wrong?

I know in Burp, it has a feature to check for validity, however, it only run 2 times then stop.
